src/python/pants/subsystem/subsystem.py

```diff
@@ -169,3 +174,33 @@ def get_options(self):
     :API: public
     """
     return self._scoped_options
+
+  @staticmethod
+  def get_streaming_workunit_callbacks(subsystem_names: List[str]) -> List[Callable]:
+    callables = []
+
+    for name in subsystem_names:
+      try:
+        module_name = '.'.join(name.split(".")[:-1])
```

Comment:
This seems reasonable, but it also seems like an operation (playing with import paths) that there might be some stdlib solution for. I think this is readable enough though, if we can add a docstring.

Reply:
Hm. In looking over the docs at https://docs.python.org/3/library/importlib.html and https://docs.python.org/3/library/functions.html#__import__, it looks like what we're doing here is reasonable enough. One thing that might be interesting is that we could do something like:
```python
module = __import__(module_name, globals(), locals(), [class_name], 0)
subsystem_class = getattr(module, class_name)
```

Basically, by using `_import__()` directly, we can do the equivalent of a `from <module_name> import <class_name>`. That distinction seems like it would only be relevant for performance reasons, and I don't think importing a single module is an optimization we need to worry about right now.

So the code as it stands is great!

Reply:
Nit: can avoid calling `name.split(".")` twice here, but that doesn't matter.